src/braket/aws/aws_session.py

```diff
@@ -157,25 +160,20 @@ def _update_user_agent(self) -> None:
         """
         Updates the `User-Agent` header forwarded by boto3 to include the braket-sdk,
         braket-schemas and the notebook instance version. The header is a string of space delimited
-        values (For example: "Boto3/1.14.43 Python/3.7.9 Botocore/1.17.44"). See:
-        https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html#botocore.config.Config
+        values (For example: "Boto3/1.14.43 Python/3.7.9 Botocore/1.17.44").
         """
 
         def _notebook_instance_version() -> str:
             # TODO: Replace with lifecycle configuration version once we have a way to access those
             nbi_metadata_path = "/opt/ml/metadata/resource-metadata.json"
             return "0" if os.path.exists(nbi_metadata_path) else "None"
 
-        additional_user_agent_fields = (
+        self._braket_user_agents = (
```

Comment:
Curious to know how did we decide to move from to self.braket_client._client_config.user_agent to _self.braket_user_agents?

Reply:
Also another silly question, I can't seem to pinpoint where the actual bugfix logic is - to my python naive eyes we simply store useragent from `self.braket_client._client_config.user_agent` to `self._braket_user_agents`. Would you mind pointing out the line that actually ensure no matter which boto version, we always add BraketSchema?

Reply:
We are now updating the user agent on each API request via a callback (`_add_braket_user_agents_handler`) registered above (`braket_client.meta.events.register(...)`). The old method of updating the user_agent string inside of `braket_client` does not work after recent changes to botocore because the new botocore implementation does not use this for creating requests.